Sample/API/API.swift

```diff
@@ -73,11 +73,13 @@ public final class MovieAPI: APIProtocol {
     
     // weaver: urlSession <- URLSession
     
-    public init(injecting dependencies: MovieAPIDependencyResolver) {
+    // weaver: logger = Logger
+    
+    init(injecting dependencies: MovieAPIDependencyResolver) {
```

Comment:
I would say that a public API should most likely a container, not a resolver.

Reply:
It should be impossible for the framework user to inject their own properties in to the API.

Reply:
The registration of dependencies is only supposed to happen when initializing the `Container` by overriding `registerDependencies(_:)`. 

Adding dependencies later on would be a very bad practice, and also prevent the main weaver promise, which is; dependency graph validation at compile time.

Reply:
I mean, shouldn't the top level API be a container? Like so:

```
public final class MovieAPI: APIProtocol {  
      private let dependencies = MovieAPI DependencyContainer()
}
```

Maybe I don't get the distinction.

Reply:
The top level API shouldn't be a container because the container allows to `resolve` & `register` dependencies, which are actions that shouldn't be exposed to the end user.

The `MyClassDependencyResolver` protocols which get generated only contain the dependencies' accessors, which is why it can be exposed to the end user.